### PR TITLE
Fix for incorrect transparent -> DividerColor animation for ExpansionTile

### PR DESCRIPTION
**Pull request by mehmetf**
Alternative fix for flutter/flutter#10464. Discussing between this and https://github.com/flutter/engine/pull/4426

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
LGTM

**Comment by Hixie** [comment]
Looks like backgroundColor has the same bug.

It might be worth also adding some commentary to ColorTween's docs, saying that rather that using an explicit color for transparent, using null will usually have a better effect and explaining why.

Also, needs a test. :-)

File: packages/flutter/test/material/expansion_tile_initial_state_test.dart [hunk 1]
```diff
@@ -72,5 +88,17 @@ void main() {
...
+    expect(expandedContainerDecoration.border.top.color, Colors.transparent);
+    expect(expandedContainerDecoration.border.bottom.color, Colors.transparent);
+
+    // Collapsed should be expanded now.
+    collapsedContainerDecoration = getContainer(collapsedKey).decoration;
+    expect(collapsedContainerDecoration.color, Colors.transparent);
+    expect(collapsedContainerDecoration.border.top.color, dividerColor);
+    expect(collapsedContainerDecoration.border.bottom.color, dividerColor);
   });
 }
```
Comment (Hixie) [inline packages/flutter/test/material/expansion_tile_initial_state_test.dart]: this test doesn't seem to check that the animation animates with the right color (i.e. that the original bug is actually fixed); am I missing something?

Comment (mehmetf) [reply packages/flutter/test/material/expansion_tile_initial_state_test.dart]: You are right. This only checks in the initial and final state to make sure my change did not introduce a regression.

I need your guidance on how to add a mid point check.

Would `pump(1/2 animation duration)` produce a stable state that can be tested?

Comment (Hixie) [reply packages/flutter/test/material/expansion_tile_initial_state_test.dart]: Yup, pumps with fixed durations will very reliably get you to the exact position in time that you want.

**Pull request merged**